Source/Core/DolphinQt/Settings/GameCubePane.cpp

```diff
@@ -53,59 +45,69 @@ GameCubePane::GameCubePane()
-    m_language_combo->addItem(language);
+  for (const auto& entry : {std::make_tuple(tr("English"), 0), std::make_tuple(tr("German"), 1),
+                            std::make_tuple(tr("French"), 2), std::make_tuple(tr("Spanish"), 3),
+                            std::make_tuple(tr("Italian"), 4), std::make_tuple(tr("Dutch"), 5)})
+  {
+    m_language_combo->addItem(std::get<0>(entry), std::get<1>(entry));
+  }
 
   ipl_layout->addWidget(m_skip_main_menu, 0, 0);
   ipl_layout->addWidget(new QLabel(tr("System Language:")), 1, 0);
   ipl_layout->addWidget(m_language_combo, 1, 1);
   ipl_layout->addWidget(m_override_language_ntsc, 2, 0);
 
   // Device Settings
-  QGroupBox* device_box = new QGroupBox(tr("Device Settings"));
-  QGridLayout* device_layout = new QGridLayout;
+  QGroupBox* device_box = new QGroupBox(tr("Device Settings"), this);
+  QGridLayout* device_layout = new QGridLayout(device_box);
   device_box->setLayout(device_layout);
 
-  m_slot_combos[0] = new QComboBox;
-  m_slot_combos[1] = new QComboBox;
-  m_slot_combos[2] = new QComboBox;
-
-  m_slot_buttons[0] = new QPushButton(tr("..."));
-  m_slot_buttons[1] = new QPushButton(tr("..."));
-
-  m_slot_buttons[0]->setSizePolicy(QSizePolicy::Fixed, QSizePolicy::Fixed);
-  m_slot_buttons[1]->setSizePolicy(QSizePolicy::Fixed, QSizePolicy::Fixed);
+  for (int i = 0; i < SLOT_COUNT; i++)
+  {
+    m_slot_combos[i] = new QComboBox(device_box);
+    m_slot_buttons[i] = new QPushButton(tr("..."), device_box);
+    m_slot_buttons[i]->setSizePolicy(QSizePolicy::Fixed, QSizePolicy::Fixed);
+  }
 
   const QString i10n_nothing = tr("<Nothing>");
   const QString i10n_dummy = tr("Dummy");
 
   // Add slot devices
 
-  for (const auto& device : {i10n_nothing, i10n_dummy, tr("Memory Card"), tr("GCI Folder"),
-                             tr("USB Gecko"), tr("Advance Game Port"), tr("Microphone")})
+  for (const auto& entry :
+       {std::make_tuple(i10n_nothing, ExpansionInterface::EXIDEVICE_NONE),
+        std::make_tuple(i10n_dummy, ExpansionInterface::EXIDEVICE_DUMMY),
+        std::make_tuple(tr("Memory Card"), ExpansionInterface::EXIDEVICE_MEMORYCARD),
+        std::make_tuple(tr("GCI Folder"), ExpansionInterface::EXIDEVICE_MEMORYCARDFOLDER),
+        std::make_tuple(tr("USB Gecko"), ExpansionInterface::EXIDEVICE_GECKO),
+        std::make_tuple(tr("Advance Game Port"), ExpansionInterface::EXIDEVICE_AGP),
+        std::make_tuple(tr("Microphone"), ExpansionInterface::EXIDEVICE_MIC)})
   {
-    m_slot_combos[0]->addItem(device);
-    m_slot_combos[1]->addItem(device);
+    m_slot_combos[0]->addItem(std::get<0>(entry), std::get<1>(entry));
+    m_slot_combos[1]->addItem(std::get<0>(entry), std::get<1>(entry));
   }
 
   // Add SP1 devices
 
-  for (const auto& device : {i10n_nothing, i10n_dummy, tr("Broadband Adapter")})
+  for (const auto& entry :
+       {std::make_tuple(i10n_nothing, ExpansionInterface::EXIDEVICE_NONE),
```

Comment:
What's i10n? :)

Reply:
Internationalization.

Reply:
Er, wait, it's supposed to be i18n, isn't it...

Reply:
I don't know, I left the name of those two const's unchanged (I just packed them into my new initializer list)

Reply:
I made them both literals, looks cleaner with all the other options.

Reply:
Interresting mix, since it's supposed to be i18n (internationalization, 18 letters) and L10n (localization, 10 letters) 😆